### PR TITLE
Unlock P12 with --pkcs12-passphrase

### PR DESCRIPTION
**Pull request by zugzwang**
Like https://github.com/fortanix/sq-dsm/pull/69 but with CLI flag instead of env variable.

### CODE REVIEW

File: openpgp-dsm/src/lib.rs [hunk 1]
```diff
@@ -1847,15 +1848,15 @@ fn api_curve_from_sequoia_curve(curve: SequoiaCurve) -> Result<ApiCurve> {
...
 
-fn try_unlock_p12(cert_file: String) -> Result<Identity> {
+fn try_unlock_p12(cert_file: String, passphrase: Option<&str>) -> Result<Identity> {
     let mut cert_stream = File::open(cert_file.clone())
         .context(format!("opening {}", cert_file))?;
     let mut cert = Vec::new();
     cert_stream.read_to_end(&mut cert)
         .context(format!("reading {}", cert_file))?;
-    // Try to unlock certificate without password first
+    // Try to parse certificate with passed or empty password first
```
Comment (zugzwang) [inline openpgp-dsm/src/lib.rs:1857]: "unlock" was the right word.

File: openpgp-dsm/src/lib.rs [hunk 2]
```diff
@@ -1847,15 +1848,15 @@ fn api_curve_from_sequoia_curve(curve: SequoiaCurve) -> Result<ApiCurve> {
...
     let mut first = true;
-    if let Ok(id) = Identity::from_pkcs12(&cert, "") {
+    if let Ok(id) = Identity::from_pkcs12(&cert, passphrase.unwrap_or("")) {
```
Comment (zugzwang) [inline openpgp-dsm/src/lib.rs:1859]: E.g. try with the passed password or the empty password, and then keep trying other things such as env variables or user prompt.

File: sq/src/sq-usage.rs [hunk 3]
```diff
@@ -104,6 +104,9 @@
 //!     -o, --output <FILE>
 //!             Writes to FILE or stdout if omitted
 //!
+//!         --pkcs12-passphrase <PKCS12-PASSPHRASE>
```
Comment (zugzwang) [inline sq/src/sq-usage.rs:107]: Auto generated file

**Review by MihirLuthra — APPROVED** [review]
Looks good to me

**Pull request merged**